src/app/_helpers/calcprogress.spec.ts

```diff
@@ -0,0 +1,31 @@
+import { WorkflowExecution } from '../_models';
+
+import { calcProgress } from './calcprogress';
+
+function makeWorkflowExecution(currIndex: number, progressStatuses: (number)[]): WorkflowExecution {
+  // tslint:disable:no-any
+  const fakePlugins: (any)[] = [];
+
+  progressStatuses.forEach((status) => {
+    fakePlugins.push({ executionProgress: { progressPercentage: status } });
+  });
+
```

Comment:
this is nicer with `map`:

    const fakePlugins = progressStatuses.map((status) => (
      { executionProgress: { progressPercentage: status } } as any
    ));

Reply:
Done